t/t7407-submodule-foreach.sh

```diff
@@ -349,11 +349,26 @@ test_expect_success 'use "git clone --recursive" to checkout all submodules' '
 	)
 '
 
+test_expect_success 'use "git clone" with submodule.recurse=true to checkout all submodules' '
+	git clone -c submodule.recurse=true super clone5 &&
+	(
+		cd clone5 &&
+		git rev-parse --resolve-git-dir .git &&
+		git rev-parse --resolve-git-dir sub1/.git &&
+		git rev-parse --resolve-git-dir sub2/.git &&
+		git rev-parse --resolve-git-dir sub3/.git &&
+		git rev-parse --resolve-git-dir nested1/.git &&
+		git rev-parse --resolve-git-dir nested1/nested2/.git &&
+		git rev-parse --resolve-git-dir nested1/nested2/nested3/.git &&
+		git rev-parse --resolve-git-dir nested1/nested2/nested3/submodule/.git
+	)
+'
+
 test_expect_success 'test "update --recursive" with a flag with spaces' '
 	git clone super "common objects" &&
-	git clone super clone5 &&
+	git clone super clone6 &&
```

Comment:
Better add the new test case at the end of the script, to avoid renaming these.

Reply:
OK, no problem. Can move it.
I thought it would be better readable to have the two tests for cloning with submodules (‘git clone –recursive’ vs ‘git clone -c submodule.recurse=true’) side by side.

Reply:
It is always a balance between readability of the code after applying the patches, and readability of the patch itself. In this case, I would prefer the latter, to have an easier time preventing unintentional mistakes (such as forgetting _one_ place to rename, or forgetting to introduce a rename when merging an unrelated branch that adds another test case, too).